__tests__/components/Send.test.js

```diff
@@ -0,0 +1,216 @@
+
+const createAsset = (name, id) => ({
+  asset: name,
+  address: '',
+  note: '',
+  amount: 0,
+  max: 10,
+  id,
+  errors: {}
+})
+
+describe('Send', () => {
+  test('It shows ZeroAssets component when no sendable assets is present', () => {
+    const wrapper = setup()
+
+    expect(wrapper.find(ZeroAssets)).toBeTruthy()
+  })
+
+  test('It adds a row when you click Add Recipient button', () => {
+    const wrapper = setup()
+
+    wrapper.find('.sendPanelHeaderButton').simulate('click')
+    expect(wrapper.instance().state.sendRowDetails.length).toBe(2)
+    expect(wrapper.find(SendRecipientListItem).children().length).toBe(2)
+  })
+
+  test('It deletes the correct row when you click the delete icon for that row', () => {
+    const wrapper = setup()
+
+    const sendRowDetails = []
+    sendRowDetails.push(
+      createAsset('NEO', 1),
+      createAsset('GAS', 2),
+      createAsset('RPX', 3)
+    )
+
+    wrapper.setState({ sendRowDetails })
+    wrapper
+      .find('.deleteButton')
+      .first()
+      .simulate('click')
+
+    const wrapperRows = wrapper.instance().state.sendRowDetails
+
+    expect(wrapperRows.length).toBe(2)
+    expect(wrapperRows[0].asset).toBe('GAS')
+    expect(wrapperRows[1].asset).toBe('RPX')
+  })
+
+  test('It adds a maximum of 5 rows', () => {
+    const wrapper = setup()
+
+    for (let i = 0; i <= 6; i++) {
+      wrapper.find('.sendPanelHeaderButton').simulate('click')
+    }
+
+    expect(wrapper.instance().state.sendRowDetails.length).toBe(5)
+    expect(wrapper.find(SendRecipientListItem).children().length).toBe(5)
+  })
+
+  test("It doesn't delete the first row", () => {
+    const wrapper = setup()
+
+    wrapper.find('.deleteButton').simulate('click')
+
+    expect(wrapper.instance().state.sendRowDetails.length).toBe(1)
+  })
+
+  test('It validates and sets error in state when you submit empty inputs', () => {
+    const wrapper = setup()
+
+    wrapper.find('form').simulate('submit')
+    const errors = wrapper.instance().state.sendRowDetails[0].errors
+
+    expect(errors.amount).toBe('Can not send 0 NEO.')
+    expect(errors.address).toBe('You need to specify a valid NEO address.')
+  })
+
+  test('It does not allow negative amounts', () => {
+    const wrapper = setup()
+
+    const asset = createAsset('NEO', 1)
+    asset.amount = -5
+    wrapper.setState({ sendRowDetails: [asset] })
+    wrapper.find('form').simulate('submit')
+
+    const errors = wrapper.instance().state.sendRowDetails[0].errors
+    expect(errors.amount).toBe('You cannot send negative amounts of NEO.')
+  })
+
+  test('It does not allow fractional amounts of NEO', () => {
+    const wrapper = setup()
+
+    const asset = createAsset('NEO', 1)
+    asset.amount = 1.5
+    wrapper.setState({ sendRowDetails: [asset] })
+    wrapper.find('form').simulate('submit')
+
+    const errors = wrapper.instance().state.sendRowDetails[0].errors
+    expect(errors.amount).toBe('You cannot send fractional amounts of NEO.')
+  })
+
```

Comment:
These tests look great man! Can we add one more verifying that an error state occurs user attempts to send assets to themself